axelor-account/src/main/java/com/axelor/apps/account/service/batch/BatchBlockCustomersWithLatePayments.java

```diff
@@ -50,47 +52,60 @@ protected void process() {
 
   protected String blockCustomersWithLatePayments() {
     StringBuilder result = new StringBuilder();
-    List<Invoice> invoices;
-    List<Long> customersToBlock = new ArrayList<>();
+    List<DebtRecovery> debtRecoveries;
+    List<Long> customersToBlock = new ArrayList<Long>();
+    List<Long> customerToUnblock = new ArrayList<Long>();
     int offset = 0;
-    Query<Invoice> query =
-        invoiceRepository
-            .all()
-            .filter(
-                "self.operationTypeSelect = :operationTypeSelect "
-                    + "AND self.amountRemaining > 0");
-    while (!(invoices =
-            query
-                .bind("operationTypeSelect", InvoiceRepository.OPERATION_TYPE_CLIENT_SALE)
-                .fetch(FETCH_LIMIT, offset))
-        .isEmpty()) {
-
-      for (Invoice invoice : invoices) {
+    Query<DebtRecovery> query =
+        debtRecoveryRepository.all().filter("self.archived = false or self.archived is null");
+    while (!(debtRecoveries = query.fetch(FETCH_LIMIT, offset)).isEmpty()) {
+      for (DebtRecovery debtRecovery : debtRecoveries) {
         ++offset;
-        try {
-          Partner partner = processInvoice(invoice);
-          if (partner != null && !customersToBlock.contains(partner.getId())) {
-            log.debug("Blocking {}", partner.getFullName());
-            result
-                .append(I18n.get("Blocking"))
-                .append(" ")
-                .append(partner.getFullName())
-                .append("</br>");
-            customersToBlock.add(partner.getId());
-            incrementDone();
+        if (debtRecovery
+                .getRespiteDateBeforeAccountBlocking()
+                .compareTo(appBaseService.getTodayDate(debtRecovery.getCompany()))
+            >= 1) {
```

Comment:
isn't it
```suggestion
            >= 0) {
```
or do we block user when we are at the respire date ? if so please change it to
```suggestion
            > 0) {
```